File: hash_table/0961_n_repeated_element_in_size_2n_array/0961_n_repeated_element_in_size_2n_array.py
from typing import List

class Solution:
    def repeatedNTimes(self, A: List[int]) -> int:
        for k in range(1, 4):
            for i in range(len(A) - k):
                if A[i] == A[i+k]:
                    return A[i]


A = [1,2,3,3]
res = Solution().repeatedNTimes(A)
print(res)

File: hash_table/0961_n_repeated_element_in_size_2n_array/0961_n_repeated_element_in_size_2n_array.go
package main

import "fmt"

func repeatedNTimes(A []int) int {
	for i := 0; i < len(A)-2; i++ {
		if A[i] == A[i+1] || A[i] == A[i+2] {
			return A[i]
		}
	}

	return A[len(A)-1]
}

func main() {
	A := []int{1, 2, 3, 3}
	res := repeatedNTimes(A)
	fmt.Println(res)
}
